components/mediation-admin/org.wso2.carbon.endpoint/src/main/java/org/wso2/carbon/endpoint/service/EndpointAdmin.java

```diff
@@ -156,15 +156,19 @@ public boolean addEndpoint(String epString) throws EndpointAdminException {
                     }
                     SynapseXMLConfigurationFactory.defineEndpoint(
                             config, endpointElement, config.getProperties());
+
                     Endpoint endpoint = config.getEndpoint(endpointName);
-                    if (endpoint != null) {
-                        if (endpoint instanceof AbstractEndpoint) {
-                            endpoint.setFileName(
-                                    ServiceBusUtils.generateFileName(endpoint.getName()));
-                        }
-                        endpoint.init(getSynapseEnvironment());
-                        persistEndpoint(endpoint);
+                    if (endpoint == null) {
+                        handleFault("Unable to create endpoint", null);
                     }
+
```

Comment:
@prabathariyaratna Is there any specific reason to add a separate if condition rather than using the else statement of the previous check?